### PR TITLE
Separating Code Generation Step 5: Amplication Server - Updating and logging build status

### PR DESCRIPTION
**Pull request by EugeneTseitlin**
Closes: #3084

## PR Details

Added to server logic that handles build status update. Added status and run id to build table.

## PR Checklist
- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/code_of_conduct.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-build-manager-service/src/app.controller.ts [hunk 1]
```diff
@@ -43,7 +43,11 @@ export class AppController {
     try {
       const path = await this.buildContextStorage.saveBuildContextSource(gr);
       const runResponse = await this.buildService.runBuild(path);
-      this.emitInitMessage(runResponse.runId, 'Generating code');
+      this.emitInitMessage(
+        gr.buildId,
```
Comment (arielweinberger) [inline packages/amplication-build-manager-service/src/app.controller.ts:47]: I think `gr` can be `generatedResource`. Let's not be cryptic.

File: packages/amplication-server/src/core/build/build.controller.ts [hunk 2]
```diff
@@ -16,17 +18,25 @@ import { StepNotCompleteError } from './errors/StepNotCompleteError';
 import { StepNotFoundError } from './errors/StepNotFoundError';
 import { CanUserAccessArgs } from './dto/CanUserAccessArgs';
 import { plainToInstance } from 'class-transformer';
-import { MessagePattern, Payload } from '@nestjs/microservices';
-import { CHECK_USER_ACCESS_TOPIC } from 'src/constants';
+import { EventPattern, MessagePattern, Payload } from '@nestjs/microservices';
+import { BUILD_STATUS_TOPIC, CHECK_USER_ACCESS_TOPIC } from 'src/constants';
```
Comment (arielweinberger) [inline packages/amplication-server/src/core/build/build.controller.ts:22]: Let's get rid of `src/` imports and make it relative

File: packages/amplication-server/src/core/build/build.service.ts [hunk 3]
```diff
@@ -236,6 +237,24 @@ export class BuildService {
...
 
+  async findByRunId(runId: string): Promise<Build | null> {
+    return this.prisma.build.findFirst({
+      where: {
+        runId
+      }
+    });
+  }
+
+  async logGenerateStatusByRunId(runId: string, status: string): Promise<void> {
```
Comment (arielweinberger) [inline packages/amplication-server/src/core/build/build.service.ts:248]: Let's change the `status: string` type to match an enum for clarity

File: packages/amplication-server/src/core/build/build.service.ts [hunk 4]
```diff
@@ -236,6 +237,24 @@ export class BuildService {
...
+    const build = await this.findByRunId(runId);
+    const steps = await this.actionService.getSteps(build.actionId)
+    const generateStep = steps.find(step => step.name === GENERATE_STEP_NAME);
```
Comment (arielweinberger) [inline packages/amplication-server/src/core/build/build.service.ts:251]: I see we are comparing `step.name` to a constant. I also see we have 3 step names defined in this file (GENERATE_APPLICATION, BUILD_DOCKER, PUSH_TO_GITHUB).

Can we please create an Enum to contain these types? For example, `CodeGenerationSteps`?

**Review by arielweinberger — CHANGES_REQUESTED** [review]

File: packages/amplication-build-manager-service/src/app.controller.ts (hunk 1 above)
Comment (EugeneTseitlin) [reply packages/amplication-build-manager-service/src/app.controller.ts:47]: Done

File: packages/amplication-server/src/core/build/build.controller.ts (hunk 2 above)
Comment (EugeneTseitlin) [reply packages/amplication-server/src/core/build/build.controller.ts:22]: change import to relative

File: packages/amplication-server/src/core/build/build.service.ts (hunk 3 above)
Comment (EugeneTseitlin) [reply packages/amplication-server/src/core/build/build.service.ts:248]: added type

File: packages/amplication-server/src/core/build/build.service.ts (hunk 4 above)
Comment (EugeneTseitlin) [reply packages/amplication-server/src/core/build/build.service.ts:251]: After checking required changes in order to transfer usages to enum, I've decided to open an issue and solve it separately.

**Pull request merged**